maplibre_gl/lib/src/controller.dart

```diff
@@ -10,13 +10,17 @@ typedef OnMapClickCallback = void Function(
 typedef OnFeatureInteractionCallback = void Function(
     dynamic id, Point<double> point, LatLng coordinates, String layerId);
 
-typedef OnFeatureDragnCallback = void Function(dynamic id,
-    {required Point<double> point,
+typedef OnFeatureDragCallback = void Function(dynamic id,
+    {required Annotation annotation,
+    required Point<double> point,
     required LatLng origin,
     required LatLng current,
     required LatLng delta,
     required DragEventType eventType});
 
```

Comment:
Since we are making a breaking change from the next version, introducing the `annotation` field in the callback, why not remove the `id?`
The developer can find the `id` into the `Annotation` fields.
```suggestion
typedef OnFeatureDragCallback = void Function({
  required Annotation annotation,
  required Point<double> point,
  required LatLng origin,
  required LatLng current,
  required LatLng delta,
  required DragEventType eventType,
});
```

Reply:
Good point. Should I switch id to annotation in OnFeatureInteractionCallback for consistency, or keep it as is to prevent a breaking change?

Reply:
We could do a breaking change there too, no problem. It is more correct that a “Feature” callback should have the annotation, for better handling of the same

Reply:
I just remembered why I didn’t remove the id from the callback. Since id is dynamic, replacing it directly with an annotation wouldn’t trigger any compile-time errors, so developers wouldn’t notice the breaking change.

Similarly, if we only renamed onFeatureDrag and onFeatureTapped, developers might just update the names without realizing that the dynamic id was swapped out for an annotation. That’s why adding the annotation as an additional parameter felt like the safest approach.

What do you think?

Reply:
I think we can still remove the `dynamic id` and, instead of replacing it with the annotation, keep the `Annotation annotation` as a named parameter. This will cause compilation errors because a required named parameter has not been declared (or am I wrong?)
What do you think about doing it this way?

Reply:
This is a good idea. 
However, I rearranged the parameters for OnFeatureInteractionCallback to cause a compile error and changed the named parameters in OnFeatureDragCallback to positional ones to maintain consistency.

Reply:
I did a quick cleanup too, removed annotationConsumeTapEvents and enableInteraction since they weren’t used. Please have another look and let me know if you spot anything else.

Reply:
I will check today, but it seems strange to me that enableInteraction is not used (I currently use it in my prod app)